index.js

```diff
@@ -0,0 +1,86 @@
+import Config from 'react-native-config'
+import * as Keychain from 'react-native-keychain'
+import { Linking } from 'react-native'
+
+const AUTH = Config.ACCOUNTS_URL + '/o/oauth2/v1/auth'
+const TOKEN = Config.ACCOUNTS_URL + '/o/oauth2/v1/token'
+const USERINFO = Config.ACCOUNTS_URL + '/o/oauth2/v1/userInfo'
+const LOGOUT = Config.ACCOUNTS_LOGOUT_URL
+const clientId = Config.APP_ID
+const callback = Config.APP_CALLBACK
+
+const authorizationUrl = 
+  (url, appId, callback, scope, responseType = 'token') =>
+    `${url}?scope=${encodeURIComponent(scope)}&
+      redirect_uri=${encodeURIComponent(callback)}&
+      response_type=${responseType}&
+      client_id=${appId}`.replace(/\s+/g, '')
+
+export default class {
+  static authenticate() {
+    const handleUrl = event => {
+      const authCode = event.url.substring(
+        event.url.indexOf('=') + 1,
+        event.url.length
+      )
+      const tokenRequest = {
+        code: authCode,
+        client_id: clientId,
+        redirect_uri: callback,
+        grant_type: 'authorization_code'
+      }
+      let s = []
+      for (let key in tokenRequest) {
+        if (tokenRequest.hasOwnProperty(key)) {
+          s.push(
+             encodeURIComponent(key) +
+              '=' +
+              encodeURIComponent(tokenRequest[key])
+          )
+        }
+      }
+      const targetUrl = s.join('&')
+      console.warn('targetUrl: ', targetUrl)
+      fetch(TOKEN, {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/x-www-form-urlencoded'
+        },
+        body: s.join('&')
+      })
+        .then(response => {
+          loggedIn = true
+        })
+        .catch(error => {
+          console.log('', error)
+        })
+      Linking.removeEventListener('url', handleUrl)
+    }
+    Linking.addEventListener('url', handleUrl)
+    Linking.openURL(authorizationUrl(AUTH, clientId, callback, 'code'))
+
+    if (Config.ENV === 'staging') {
+      // ToDo: Fetching User Info
+    } else if (process.env.NODE_ENV === 'development') {
+      return {
+        userInfo: {
+          id: '5735995932672001',
+          first_name: 'John',
+          family_name: 'Doe'
+        },
+        credentials: {}
+      }
+    }
+  }
+
+  static logout() {
+    Linking.openURL(LOGOUT + callback)
+      .then(response => {
+        return false
```

Comment:
return is not required here as it's in a promise

Reply:
fixed